genesyscloud/architect_user_prompt/resource_architect_user_prompt_schema.go

```diff
@@ -0,0 +1,750 @@
+package architect_user_prompt
+
+import (
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	registrar "terraform-provider-genesyscloud/genesyscloud/resource_register"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+const resource_name = "genesyscloud_architect_user_prompt"
+
+// SetRegistrar registers all of the resources, datasources and exporters in the package
+func SetRegistrar(regInstance registrar.Registrar) {
+	regInstance.RegisterResource(resource_name, ResourceArchitectUserPrompt())
+	regInstance.RegisterDataSource(resource_name, DataSourceUserPrompt())
+	regInstance.RegisterExporter(resource_name, ArchitectUserPromptExporter())
+}
+func ArchitectUserPromptExporter() *resourceExporter.ResourceExporter {
+	return &resourceExporter.ResourceExporter{
+		GetResourcesFunc: provider.GetAllWithPooledClient(getAllUserPrompts),
+		RefAttrs:         map[string]*resourceExporter.RefAttrSettings{}, // No references
+		CustomFileWriter: resourceExporter.CustomFileWriterSettings{
+			RetrieveAndWriteFilesFunc: ArchitectPromptAudioResolver,
+			SubDirectory:              "audio_prompts",
+		},
+	}
+}
+
+func DataSourceUserPrompt() *schema.Resource {
+	return &schema.Resource{
+		Description: "Data source for Genesys Cloud User Prompts. Select a user prompt by name.",
+		ReadContext: provider.ReadWithPooledClient(dataSourceUserPromptRead),
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "User Prompt name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+var userPromptResource = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"language": {
+			Description: "Language for the prompt resource. (eg. en-us)",
+			Type:        schema.TypeString,
+			Required:    true,
+			ValidateFunc: validation.StringInSlice([]string{
```

Comment:
Instead of naming all these values there is a constant used in architect grammar you can use here